File: thirtysix.py
#!/usr/bin/env python3

def is_palindrome(word):
   for i in range(len(word) // 2):
         if word[i] != word[-1 - i]:
             return False
   return True


def find_double_base_palindromes_to(n):
    a = []

    for i in range(n):
        if is_palindrome(str(i)) and is_palindrome(bin(i)[2:]):
            a.append(i)
        
    return a


if __name__ == '__main__':
    a = find_double_base_palindromes_to(1000000)
    print(a)
    print(sum(a))    

File: thirtysix.hs
import Numeric (showIntAtBase)
import Data.Char (intToDigit)

isPalindrome :: (Eq a) => [a] -> Bool
isPalindrome xs = xs == (reverse xs)

intToBin :: (Integral a, Show a) => a -> String
intToBin n = showIntAtBase 2 intToDigit n ""

findDoubleBasePalindromesTo :: Int -> [Int]
findDoubleBasePalindromesTo 0 = []
findDoubleBasePalindromesTo n = if (
  isPalindrome (show n) && isPalindrome (intToBin n))
  then n : (findDoubleBasePalindromesTo (n-1))
  else (findDoubleBasePalindromesTo (n-1))

main = putStrLn $ show $ sum $ findDoubleBasePalindromesTo 999999
